db/migrate/20171222120314_add_constraints_on_discovery_rules_hostgroups.rb

```diff
@@ -0,0 +1,8 @@
+ class AddConstraintsOnDiscoveryRulesHostgroups < ActiveRecord::Migration[4.2]
+  
+   def change
+    DiscoveryRule.unscoped.where("hostgroup_id IS NULL OR hostgroup_id NOT IN (?)", Hostgroup.unscoped.all.pluck(:id)).delete_all
```

Comment:
Damn one more thing, this should be written as:

```ruby
      reversible do |dir|
        dir.up do
          DiscoveryRule.unscoped.where("hostgroup_id IS NULL OR hostgroup_id NOT IN (?)", Hostgroup.unscoped.all.pluck(:id)).    +delete_all
        end
      end
```

I made the change in separate commit: 6efc6cc